File: .tmp_py4013/batch_09_dump/3021_alice_and_bob_playing_flower_game.js
// LeetCode 3021 - Alice and Bob Playing Flower Game
// https://leetcode.com/problems/alice-and-bob-playing-flower-game/

var flowerGame = function(n, m) {
    const a1 = ((n + 1) / 2) | 0, b1 = ((m + 1) / 2) | 0;
    const a2 = (n / 2) | 0, b2 = (m / 2) | 0;
    return a1 * b2 + a2 * b1;
};

File: .tmp_rs4013/rust_check13/3021_alice_and_bob_playing_flower_game.rs
#![allow(dead_code, unused_variables, unused_mut, unused_assignments, unused_imports)]
struct Solution;
// LeetCode 3021 - Alice and Bob Playing Flower Game
// https://leetcode.com/problems/alice-and-bob-playing-flower-game/

impl Solution {
    pub fn flower_game(n: i32, m: i32) -> i64 {
        let a1 = (n as i64 + 1) / 2;
        let b1 = (m as i64 + 1) / 2;
        let a2 = n as i64 / 2;
        let b2 = m as i64 / 2;
        a1 * b2 + a2 * b1
    }
}
